libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/map/widgets/compass/CarCompassSurfaceRenderer.kt

```diff
@@ -20,13 +23,14 @@ class CarCompassSurfaceRenderer(
             compassWidget?.updateBearing(it)
         }
     }
+    private var styleLoadedListener: OnStyleLoadedListener? = null
 
     override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
         val compassWidget = CompassWidget(mapboxCarMapSurface.carContext)
         val mapboxMap = mapboxCarMapSurface.mapSurface.getMapboxMap().also { mapboxMap = it }
         this.compassWidget = compassWidget
-        mapboxMap.getStyle { style ->
-            style.addPersistentStyleCustomLayer(
+        styleLoadedListener = mapboxCarMapSurface.handleStyleOnAttached {
+            it.addPersistentStyleCustomLayer(
```

Comment:
is it an issue to add a persistent style custom layer multiple times? 

another note is that, we want to delete this class and adopt the new map widgets https://github.com/mapbox/mapbox-navigation-android/issues/5793.

The solution here will be temporary, but the solution would translate to other persistent styles

Reply:
added a flag

Reply:
It won't crash but will return `Expected.error` from 
`Expected<String, com.mapbox.bindgen.None> addPersistentStyleCustomLayer(@NonNull String layerId, @NonNull CustomLayerHost layerHost, @Nullable LayerPosition layerPosition);`. 
It's not a big issue but I assume still better to fix to avoid possible issues.